pkg/model/resource/gvk_test.go

```diff
@@ -47,13 +50,11 @@ var _ = Describe("GVK", func() {
 			Entry("Domain (non-alpha characters)", GVK{Group: group, Domain: "_*?", Version: version, Kind: kind}),
 			Entry("Group and Domain (empty)", GVK{Group: "", Domain: "", Version: version, Kind: kind}),
 			Entry("Version (empty)", GVK{Group: group, Domain: domain, Version: "", Kind: kind}),
-			Entry("Version (no v prefix)", GVK{Group: group, Domain: domain, Version: "1", Kind: kind}),
-			Entry("Version (wrong prefix)", GVK{Group: group, Domain: domain, Version: "a1", Kind: kind}),
-			Entry("Version (unstable no v prefix)", GVK{Group: group, Domain: domain, Version: "1beta1", Kind: kind}),
-			Entry("Version (unstable no alpha/beta number)",
-				GVK{Group: group, Domain: domain, Version: "v1beta", Kind: kind}),
-			Entry("Version (multiple unstable)",
-				GVK{Group: group, Domain: domain, Version: "v1beta1alpha1", Kind: kind}),
+			Entry("Version (wrong prefix)", GVK{Group: group, Domain: domain, Version: "-example.com", Kind: kind}),
```

Comment:
Hi @Kavinjsir,

Thank you so much for looking into this issue.

To ensure that the previously allowed values continue to function as expected, it's crucial that we maintain all existing tests and add new ones to verify the recent changes.

Could you please review the updates and confirm their impact? Additionally, if necessary, please revert the changes to maintain the integrity of our project.

Thank you for your attention to this matter.

Best regards,